app/gui/view/component-browser/searcher-list-panel/src/lib.rs

```diff
@@ -0,0 +1,509 @@
+    pub label:   text::Area,
+    pub divider: hline::View,
+    pub content: T,
+}
+
+type WideSection = LabeledSection<component_group::wide::View>;
+type ColumnSection = LabeledSection<column_grid::ColumnGrid>;
+
+impl<T: CloneRef> LabeledSection<T> {
+    pub fn new(content: T, app: &Application) -> Self {
+        let logger = Logger::new("LabeledSection");
+        let label = text::Area::new(app);
+        let divider = hline::View::new(logger);
+        divider.size.set(Vector2(INFINITE, SECTION_DIVIDER_HEIGHT));
+
+        Self { label, divider, content }.init_label()
+    }
+
+    fn init_label(self) -> Self {
+        let label = &self.label;
+        let section_header_color: color::Rgba =
+            color::Rgb::from_css_hex(SECTION_HEADING_COLOR_HEX).unwrap().into();
+        label.set_default_color(section_header_color);
+        label.set_default_text_size(SECTION_HEADING_SIZE);
+        label.set_font(SECTION_HEADING_FONT.to_string());
+        label.set_position_y(-0.75 * SECTION_HEADING_SIZE.raw);
+        label.set_position_x(3.0 + PADDING_INNER);
+        self
+    }
+
+    /// Returns the size of the header, which consists of the header text and the padding between
+    /// text and content.
+    fn header_height(&self) -> f32 {
+        let label_height = SECTION_HEADING_SIZE.raw;
+        label_height + SECTION_HEADER_OFFSET
+    }
+}
+
+impl<T: ObjectOps + CloneRef> LabeledSection<T> {
+    fn set_parent(&self, parent: impl ObjectOps) {
+        parent.add_child(&self.content);
+        parent.add_child(&self.label);
+        parent.add_child(&self.divider);
+    }
+}
+
+/// Trait that provides functionality for layouting and layer setting for structs used in the
+/// `LabeledSection`.
+trait SectionContent {
+    fn set_layers(&self, layers: &Layers, scroll_layer: &Layer);
+    fn height(&self) -> f32;
+    fn set_position_top_y(&self, position_y: f32);
+}
+
+impl SectionContent for component_group::wide::View {
+    fn set_layers(&self, layers: &Layers, _scroll_layer: &Layer) {
+        self.model().set_layers(layers);
+    }
+
+    fn height(&self) -> f32 {
+        self.size.value().y
+    }
+
+    fn set_position_top_y(&self, position_y: f32) {
+        self.set_position_y(position_y - self.height() / 2.0);
+    }
+}
+
+impl SectionContent for column_grid::ColumnGrid {
+    fn set_layers(&self, layers: &Layers, scroll_layer: &Layer) {
+        self.model().set_layers(layers, scroll_layer);
+    }
+
+    fn height(&self) -> f32 {
+        self.size.value().y
+    }
+
+    fn set_position_top_y(&self, position_y: f32) {
+        self.set_position_y(position_y);
+    }
+}
+
+impl<T: SectionContent + CloneRef> LabeledSection<T> {
+    fn set_layers(&self, layers: &Layers, scroll_layer: &Layer) {
+        self.content.set_layers(layers, scroll_layer);
+
+        scroll_layer.add_exclusive(&self.label);
+        self.label.add_to_scene_layer(scroll_layer);
+
+        scroll_layer.add_exclusive(&self.divider);
```

Comment:
Why these empty lines? If you feel that these 2 lines do sometghing that is worth separating, refactor that to a well named function instead, please.